pkg/controllers/templating_reconciler.go

```diff
@@ -125,6 +143,7 @@ type TemplatingReconciler struct {
 	childResourcePatcher resource.ChildResourcePatcherChain
 }
 
+// Reconcile is called by controller-runtime for reconciliation.
 func (r *TemplatingReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), reconcileTimeout)
 	defer cancel()
```

Comment:
If we want this changeset to close https://github.com/crossplane/crossplane/issues/1203, it seems like we'll want to add some more logging. The goal is to be able to [debug the controller's behavior just by inspecting logs and events](https://github.com/crossplane/crossplane/blame/14a39bb783a18aad48711ae3a35a46b3d642e69a/docs/observability-developer-guide.md#L18). This means that, for example, every `return` from the `Reconcile` method should have a log line explaining why the reconciler is exiting (at `DEBUG` if it's not because of an error). For the first case, where the resource no longer exists, I would consider logging something like `r.log.Debug("Ending reconcile with no requeue because the resource no longer exists", "request", req)`